lib/data/block_benchmark_test.go

```diff
@@ -0,0 +1,46 @@
+package data
+
+import (
+	"testing"
+
+	"github.com/ClickHouse/clickhouse-go/lib/column"
+)
+
+func Benchmark_BlockWriteArrayWithLen4(b *testing.B) {
+
+	block := &Block{}
+	block.NumColumns = 1
+
+	c, err := column.Factory("test", "Int32", nil)
+	if err != nil {
+		b.Fatal(err)
+	}
+	block.Columns = append(block.Columns, c)
+	block.Reserve()
+	b.ReportAllocs()
+
+	for i := 0; i < b.N; i++ {
+		block.WriteArrayLen(4, 0, 1)
+		for i := 0; i > 4; i++ {
+			block.WriteInt32(0, 1)
+		}
+	}
```

Comment:
This only works for one level arrays? Does it work for Nested Arrays as well?

Reply:
I add another benchmark